### PR TITLE
Add validations

### LINKED ISSUE #12: Add validations to Shop model
Add a `valid?` method to check the presence of
- name
- description

when the object is not valid, it should respond to a method`errors` which returns an array of error messages (simple strings).

### PR DESCRIPTION
**Pull request by mlan080**
Add 3 tests: 
1. Validate name and description 
2. Raise error if name is missing
3.  Raise error if name is not a string

close #12

### CODE REVIEW

File: app/models/shop.rb [hunk 1]
```diff
@@ -11,6 +11,12 @@ def create
 
   def initialize(name = nil)
    @name = name
+    raise 'name is nil'if @name == nil
```
Comment (lucasrenan) [inline app/models/shop.rb]: the idea here is more to validate before create/update instead of when initializing.

File: spec/shop_spec.rb [hunk 2]
```diff
@@ -70,4 +70,23 @@
       expect{ shop.delete }.to change { Shop.count }.by(-1)
     end
   end
+
+  describe "#valid?" do
+    context 'When shop is valid' do
+      let(:shop) { Shop.new(name: 'mandy') }
+      it 'valid'
```
Comment (lucasrenan) [inline spec/shop_spec.rb]: `it` expects a block `do...end`

File: spec/shop_spec.rb [hunk 3]
```diff
@@ -70,4 +70,27 @@
       expect{ shop.delete }.to change { Shop.count }.by(-1)
     end
   end
+
+  describe "#valid?" do
+    context 'When shop is valid' do
```
[the remark this answers is not in the document]
Comment (mlan080) [reply spec/shop_spec.rb]: After rspec 3.0 they've changed syntax to: 

to be_truthy 
to be_falsey 

hehe =/

File: app/models/shop.rb [hunk 4]
```diff
@@ -43,4 +44,15 @@ def update
   def delete
     DB[:shops].where(id: id).delete
   end
+
+  def valid?
+    add_error("name can't be blank") if name.to_s.empty? #array error message
```
[the remark this answers is not in the document]
Comment (mlan080) [reply app/models/shop.rb]: Is there a way to refactor the below into something shorter? 

  add_error("name can't be blank") if name.to_s.empty?
  add_error("description can't be blank") if description.to_s.empty?

**Pull request merged**